internal/provider/auth.go

```diff
@@ -270,11 +271,41 @@ func (l *AuthLoginCommon) init(d *schema.ResourceData) (string, map[string]inter
 	return path, params, nil
 }
 
-func (l *AuthLoginCommon) checkRequiredFields(d *schema.ResourceData, required ...string) error {
+type authDefault struct {
```

Comment:
Might add a comment that env vars take precedence over the default value if any are set?